File: Python/entities/product.py
from entities.furnish_type import FurnishType
from entities.product_type import ProductType


class Product:
    def __init__(self, id, floor_number, furnish_type, number_of_floors, number_of_rooms,
                 size, product_type, year_of_construction, location_id):
        self.id = id
        if isinstance(product_type, int):
            self.product_type = ProductType(product_type).name
        else:
            self.product_type = product_type
        if isinstance(furnish_type, int):
            self.furnish_type = FurnishType(furnish_type).name
        else:
            self.furnish_type = furnish_type
        self.floor_number = floor_number
        self.number_of_floors = number_of_floors
        self.size = size
        self.year_of_construction = year_of_construction
        self.location_id = location_id
        self.number_of_rooms = number_of_rooms

    # def __init__(self, id, product_type, furnish_type, floor_number, number_of_floors, size,
    #              year_of_construction, location_id, number_of_rooms):
    #     self.id = id
    #     self.product_type = product_type
    #     self.furnish_type = furnish_type
    #     self.floor_number = floor_number
    #     self.number_of_floors = number_of_floors
    #     self.size = size
    #     self.year_of_construction = year_of_construction
    #     self.location_id = location_id
    #     self.number_of_rooms = number_of_rooms

    def __eq__(self, other):
        return self.size == other.size and \
               self.product_type == other.product_type and \
               self.furnish_type == other.furnish_type and \
               self.floor_number == other.floor_number and \
               self.number_of_floors == other.number_of_floors and \
               self.year_of_construction == other.year_of_construction and \
               self.number_of_rooms == other.number_of_rooms

    def __str__(self):
        return str(self.id) + "," + str(self.size) + "," + str(self.location_id) + "," \
               + self.product_type.name.upper() + ',' + self.furnish_type.name.upper() + "," \
               + self.floor_number + "," + self.number_of_floors + "," + self.year_of_construction + "," \
               + self.number_of_rooms

File: Java/price-monitoring/src/main/java/page/classes/Product.java
package page.classes;

import entities.LocationEntity;
import lombok.Getter;
import lombok.Setter;
import repositories.location.LocationRepository;
import util.algo.find.products.DefaultProductFinder;
import util.enums.FurnishType;
import util.models.EstimatePriceModel;

import javax.annotation.PostConstruct;
import javax.enterprise.context.RequestScoped;
import javax.faces.context.FacesContext;
import javax.inject.Inject;
import javax.inject.Named;
import java.io.IOException;
import java.io.Serializable;
import java.util.List;
import java.util.Map;
import java.util.UUID;

@Named
@RequestScoped
@Getter
@Setter
public class Product implements Serializable {
    @Inject
    private LocationRepository locationRepository;
    @Inject
    private DefaultProductFinder productFinder;
    private List<LocationEntity> locationEntities;

    private Information information;

    @PostConstruct
    public void init(){
        information = new Information();
        locationEntities = locationRepository.getAll();
    }

    public void estimation(){
        information.getProductEntity().setLocation(locationRepository.getById(UUID.fromString(information.getLocation())));
        Map<String, Object> parameterValue = FacesContext.getCurrentInstance().getExternalContext().getSessionMap();
        parameterValue.put("products", productFinder.prepareQuery(information));
        try {
            FacesContext.getCurrentInstance().getExternalContext().redirect("http://localhost:8080/price_monitoring_war/pages/productsFound.xhtml");
        } catch (IOException e) {
        }
    }

    public void predict(){
        EstimatePriceModel estimatePriceModel = EstimatePriceModel.builder()
                .floor_number(information.getProductEntity().getFloorNumber())
                .location_id(UUID.fromString(information.getLocation()))
                .furnish_type(FurnishType.NULL)
                .number_of_floors(information.getProductEntity().getNumberOfFloors())
                .number_of_rooms(information.getProductEntity().getNumberOfRooms())
                .product_type(information.getProductEntity().getType())
                .size(information.getProductEntity().getSize())
                .year_of_construction(information.getProductEntity().getYearOfConstruction())
                .build();
        Map<String, Object> parameterValue = FacesContext.getCurrentInstance().getExternalContext().getSessionMap();
        parameterValue.put("product", estimatePriceModel);
        try {
            FacesContext.getCurrentInstance().getExternalContext().redirect("http://localhost:8080/price_monitoring_war/pages/priceEstimation.xhtml");
        } catch (IOException e) {
        }
    }
}
